### DISCUSSION TITLE
Deta with Nodejs

### DISCUSSION THREAD

**Discussion by Kayode-Lateef** [deta/cloud-docs#463] [General]
I am trying to deploy a nodejs application with static html,css and js files 

How do I go about it ..I ve read the docs but still not getting it right.

**Comment by Sokhavuth** [comment]
To use static files, you need to define an environment variable in .env file. Supposing that you know everything about .env file and how to use it.
```
BINARY_CONTENT_TYPES=image/*,font/*
```
In case you need to know what is .env and how to use it, here is the link to Deta document about this issue: [https://docs.deta.sh/docs/micros/env_vars](https://docs.deta.sh/docs/micros/env_vars)

**Comment by Kayode-Lateef** [comment]
Thanks..can you please share a doc to how to work with the .env

I have no ideas

**Reply by jnsougata** [reply]
Here's the link for the docs: https://docs.deta.sh/docs/cli/commands#deta-update

**Reply by Sokhavuth** [reply]
To make long story short, we can say that .env file is where we can keep all confidential information relating to our application. 

In our code, sometimes, we need to write our confidential information such as passwords and/or the link to our database for example to be able to connect to the database server. So, if we push our code to GitHub or other git hosting platform, our confidential information will be seen by other people who would be able to do anything with our database using these confidential information. 

To avoid this problem, we can create an .env file in which we can create environment variables representing our confidential information, and we use only these environment variables in our code. So nobody can see the real confidential information, they see only the environment variables if we do not push the .env file to git platform.

On Deta platform, if Deta CLI was already installed, to be able to use static files, we need to create an environment variable ```BINARY_CONTENT_TYPES=image/*,font/*``` in .env file, and write "deta update -e .env" on Terminal window to update the environment variables locally and on Deta server.

In your case, you just create a file with the name ".env" in the root folder, and write down ```BINARY_CONTENT_TYPES=image/*,font/*``` and update the environment variables by writing on Terminal window "deta update -e .env". After that, you would be able to use static files.

**Reply by Kayode-Lateef** [reply]
Thank you for this explanation.. I will try all these now
